subsys/bluetooth/host/iso.c

```diff
@@ -1065,63 +1099,117 @@ static struct bt_iso_cig *get_free_cig(void)
 	return NULL;
 }
 
-static int cig_init_cis(struct bt_iso_cig *cig)
+static bool cis_is_in_cig(const struct bt_iso_cig *cig,
+			  const struct bt_iso_chan *cis)
 {
-	for (int i = 0; i < cig->num_cis; i++) {
-		struct bt_iso_chan *cis = cig->cis[i];
-
-		CHECKIF(cis == NULL) {
-			BT_DBG("CIS was NULL");
-			return -EINVAL;
-		}
-
-		CHECKIF(!valid_chan_qos(cis->qos)) {
-			BT_DBG("Invalid QOS");
-			return -EINVAL;
-		}
+	return cig->id == cis->iso->iso.cig_id;
+}
 
-		if (cis->iso != NULL) {
-			BT_DBG("CIS conn was already allocated");
-			return -EALREADY;
-		}
+static int cig_init_cis(struct bt_iso_cig *cig,
+			const struct bt_iso_cig_param *param)
+{
+	for (int i = 0; i < param->num_cis; i++) {
+		struct bt_iso_chan *cis = param->cis_channels[i];
 
-		cis->iso = iso_new();
 		if (cis->iso == NULL) {
-			BT_ERR("Unable to allocate CIS connection");
-			return -ENOMEM;
-		}
+			cis->iso = iso_new();
+			if (cis->iso == NULL) {
+				BT_ERR("Unable to allocate CIS connection");
+				return -ENOMEM;
+			}
 
-		cis->iso->iso.cig_id = cig->id;
-		cis->iso->iso.is_bis = false;
-		cis->iso->iso.cis_id = i;
+			cis->iso->iso.cig_id = cig->id;
+			cis->iso->iso.is_bis = false;
+			cis->iso->iso.cis_id = cig->num_cis++;
```

Comment:
Dont you need a check for `cig->num_cis++` , I believe `CHECKIF` is used in the caller API (can that be compiled out?) , is it checked there to ensure there is no overflow here?

Reply:
I am not fully sure what you would check here, that num_cis is larger than `CONFIG_BT_ISO_MAX_CHAN`?

Reply:
Yes, as `cig_init_cis` is in the user call path with return value, `cig->num_cis` should be check with `CONFIG_BT_ISO_MAX_CHAN`  and `-ENOMEM` be returned if stream ids are exhausted. I could not find where `cig->num_cis` is initialised or reset, it is designed to roll-over, should it be renamed as `cis_id_next` or `cis_id_curr`  if the purpose is  to assign unique CIS id?

Reply:
If too many CIS are supplied, the above check will fail:
```
cis->iso = iso_new();
if (cis->iso == NULL) {
	BT_ERR("Unable to allocate CIS connection");
	return -ENOMEM;
}
```

In case of error, `cleanup_cig` would be called, but I see a problem with that, as we may also call `cleanup_cig` in the reconfigure function, which we shouldn't. I'll add an update for that part, but otherwise the remaining values should already have been checked.